vuepress/docs/next/tutorials/consume/external-db-ms.md

```diff
@@ -0,0 +1,203 @@
+                    "valueFrom": {
+                        "secretKeyRef": {
+                            "name": "ed1d60fc-285a480f-external-books-service-mysql-secret",
+                            "key": "SPRING_DATASOURCE_USERNAME"
+                        }
+                    }
+                },
+                {
+                    "name":"SPRING_DATASOURCE_PASSWORD",
+                    "valueFrom": {
+                        "secretKeyRef": {
+                            "name": "ed1d60fc-285a480f-external-books-service-mysql-secret",
+                            "key": "SPRING_DATASOURCE_PASSWORD"
+                        }
+                    }
+                },
+                {
+                    "name":"SPRING_DATASOURCE_URL",
+                    "valueFrom": {
+                        "secretKeyRef": {
+                            "name": "ed1d60fc-285a480f-external-books-service-mysql-secret",
+                            "key": "SPRING_DATASOURCE_URL"
+                        }
+                    }
+                },
+                {
+                    "name":"SPRING_JPA_DATABASE_PLATFORM",
+                    "value": "org.hibernate.dialect.MySQL8Dialect"
+                }
+            ]
+        }
+    ]
+    ...
+    "name": "YOUR-BUNDLE-NAME",
+    "version": "1.0.0",
+    "description": "a description",
+    "type": "bundle"
+}
+```
+3. Pack and publish your bundle:
+```
+  ent bundle pack
+  ent bundle publish
+```
+
+### Create and Apply a Plugin Configuration Secret
+Prior to installing a bundle, a plugin configuration Secret must be created to provide a different datasource for each tenant. Create a Secret for each microservice in the bundle to provide a series of values required by the Spring Dynamic Datasource Routing for every tenant as needed.  
+
+1. Retrieve the bundle ID and the plugin code:
+ ```
+ ent ecr get-bundle-id docker://registry.hub.docker.com/YOUR-ORG/YOUR-BUNDLE
+ ```
+ ```
+ ent ecr get-plugin-code YOUR-ORG/YOUR-MS-NAME --repo=docker://registry.hub.docker.com/YOUR-ORG/YOUR-BUNDLE
+ ```
+ 
+2. Create the JSON plugin configuration Secret for the microservice plugin with a name that reflects the bundle ID and plugin code as Kuternetes is not tenant aware.
+
+   The name should begin with "pn" plus the concatenation of the bundle ID, plugin code, and microservice name, connected by dashes. The name can be appended with suffixes such as "SECRET" for better identification.  
+
+   E.g.  
+   Bundle ID = `hasldk12`, Plugin Code = `8dsjahj2`, Microservice name = YOUR-MS-NAME  
+   Your Configuration Secret= `pn-hasldk12-8dsjahj2-YOUR-MS-NAME-SECRET`
+
+3. Add the following env vars per tenant. The key for your primary tenant must be 'primary' regardless of what the actual tenant ID is. Secondary tenant keys should refer to their tenant IDs. Add the key-value pairs for the Spring Datasource Username, password, URL, and connection driver, in a base 64 encoded format.
```

Comment:
Here we must talk about tenant code, because the tenant ID is the result of hashing the tenant code.
E.g. 
- `tenant1` => tenant code
- `a1b2a3c4` => tenant ID obtained by hashing the tenant code

Reply:
Ok understood. There is a bit of a mess around these terms. Anyway, since 2 lines up we are explicitly talking about Bundle ID to refer to a hash, can we provide an example of what a tenant ID is? Just to be clearer

Reply:
1. I don't mind changing the reference to TENANT-NAME, especially if at some point in the future we are going to have to reference the 'tenant ID' you mentioned which might happen. I think it's better to resolve these things at the earliest possible moment. But I just need to make sure that it makes sense overall, in the bigger picture, to call it a TENANT-NAME, if that is the better solution. So...
we can keep it YOUR-TENANT-ID with additional explanation or switch it now to YOUR-TENANT-NAME if that makes more sense overall. 
2. The bundle ID is removed from the the whole tutorial at any rate.